example/node-pools/main.tf

```diff
@@ -0,0 +1,152 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=2.48.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "=2.3.0"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "=1.13.0"
+    }
+    helm = {
+      source  = "hashicorp/helm"
+      version = "=1.3.2"
+    }
+  }
+  required_version = "=0.14.7"
+}
+
+provider "azurerm" {
+  features {}
+}
+
+provider "kubernetes" {
+  host                   = module.kubernetes.host
+  client_certificate     = base64decode(module.kubernetes.client_certificate)
+  client_key             = base64decode(module.kubernetes.client_key)
+  cluster_ca_certificate = base64decode(module.kubernetes.cluster_ca_certificate)
+  load_config_file       = false
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.kubernetes.host
+    client_certificate     = base64decode(module.kubernetes.client_certificate)
+    client_key             = base64decode(module.kubernetes.client_key)
+    cluster_ca_certificate = base64decode(module.kubernetes.cluster_ca_certificate)
+  }
+}
+
+data "http" "my_ip" {
+  url = "http://ipv4.icanhazip.com"
+}
+
+data "azurerm_subscription" "current" {
+}
+
+resource "random_string" "random" {
+  length  = 12
+  upper   = false
+  special = false
+}
+
+resource "random_password" "admin" {
+  length      = 14
+  special     = true
+}
+
+module "subscription" {
+  source = "github.com/Azure-Terraform/terraform-azurerm-subscription-data.git?ref=v1.0.0"
+  subscription_id = data.azurerm_subscription.current.subscription_id
+}
+
+module "naming" {
+  source = "github.com/Azure-Terraform/example-naming-template.git?ref=v1.0.0"
+}
+
+module "metadata" {
+  source = "github.com/Azure-Terraform/terraform-azurerm-metadata.git?ref=v1.1.0"
```

Comment:
We have a v1.5.0 now
Not sure if it has anything of use for you - but might be worth updating if you're already editing the file

Reply:
good catch